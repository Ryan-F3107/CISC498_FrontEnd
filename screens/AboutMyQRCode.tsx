import React from 'react';
import { Text, View, TouchableOpacity, Image, TextInput } from 'react-native';
import { IconButton } from 'react-native-paper';
//import * as Print from 'expo-print';
import styles from '../styling/styles';

function AboutMyQRCode({ navigation }) {
    const v = "ABC company"
    return (
        <View style={styles.homeContainer}>
            <IconButton
                style={styles.closeButton}
                icon="close"
                size={35}
                color={'black'}
                onPress={() => { navigation.replace("ViewMyQRCode") }}
            ></IconButton>
            <Text style={{ alignSelf: 'center', fontSize: 30, paddingBottom: 10 }}> Preview </Text>
            <View style={styles.EditQRCodeContainer}>
                <TextInput
                    placeholder={v}
                    style={{ alignSelf: 'center', marginTop: 20, fontSize: 30, paddingBottom: 10 }} />
                <Image
                    //Will be an image of a poster
                    //uri 
                    source={require('../logo/logoPlaceholder.png')}
                    style={{ alignSelf: 'center', width: 250, height: 250, marginTop: 50, marginBottom: 100, borderWidth: 1, resizeMode: 'contain' }} />
                <TextInput
                    placeholder="Edit Message: Scan the QR code to check into our business."
                    multiline={true}
                    numberOfLines={5}
                    style={{ alignSelf: 'center', borderBottomWidth: 1, width: 300, fontSize: 20, paddingBottom: 10 }} />
            </View >

            <TouchableOpacity
                style={styles.ViewQRCodebutton}
                onPress={() => navigation.replace("ViewMyQRCode")}
            >
                <Text style={{ color: '#fafafa', alignSelf: 'center' }}>Save</Text>
            </TouchableOpacity>


        </View >
    )
}
export default AboutMyQRCode;