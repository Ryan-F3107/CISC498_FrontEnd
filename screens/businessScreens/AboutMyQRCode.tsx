import React from 'react';
import { Text, View, TouchableOpacity, Image, TextInput } from 'react-native';
import { IconButton } from 'react-native-paper';
//import * as Print from 'expo-print';
import QRCode from 'react-native-qrcode-svg';
import styles from '../../styling/styles';

function AboutMyQRCode({ navigation }) {
    const v = "ABC company"
    return (
        <View style={styles.homeContainer}>
            <IconButton
                style={styles.closeButton}
                icon="close"
                size={35}
                color={'black'}
                onPress={() => { navigation.goBack() }}
            ></IconButton>
            <Text style={{ alignSelf: 'center', fontSize: 30, paddingBottom: 10 }}> Preview </Text>
            <View style={styles.EditQRCodeContainer}>
                {/*<TextInput
                    //placeholder={v}
                style={{ alignSelf: 'center', marginTop: 20, fontSize: 30, paddingBottom: 10 }} />*/}

                <View
                    style={{ alignSelf: 'center', marginTop: 10, marginBottom: 100 }}>
                    <QRCode
                        value="example@example.com"
                        size={200}
                        logoBackgroundColor='transparent'
                    />
                </View>

                {/*<TextInput
                    //placeholder="Edit Message: Scan the QR code to check into our business."
                    //borderBottomWidth: 1,
                    multiline={true}
                    numberOfLines={5}
                style={{ alignSelf: 'center', width: 300, fontSize: 20, paddingBottom: 10 }} />*/}
            </View >

            {/*<TouchableOpacity
                style={styles.ViewQRCodebutton}
                onPress={() => navigation.replace("ViewMyQRCode")}
            >
                <Text style={{ color: '#fafafa', alignSelf: 'center' }}>Save</Text>
            </TouchableOpacity>*/}


        </View >
    )
}
export default AboutMyQRCode;