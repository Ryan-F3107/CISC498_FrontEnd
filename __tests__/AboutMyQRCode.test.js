import React from 'react';
import renderer from 'react-test-renderer';
import AboutMyQRCode from '../screens/businessScreens/AboutMyQRCode';

describe("<AboutMyQRCode />", () => {
    it("has 4 children", () => {
        const tree = renderer.create(<AboutMyQRCode />).toJSON();
        expect(tree.children.length).toBe(4);
    });

    it('renders correctly', () => {
        renderer.create(<AboutMyQRCode />);
    });

    it('AboutMyQRCode testing with snapshot', () => {
        const tree = renderer.create(<AboutMyQRCode />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});