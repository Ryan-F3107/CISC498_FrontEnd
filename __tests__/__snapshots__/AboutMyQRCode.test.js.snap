// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutMyQRCode /> AboutMyQRCode testing with snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fafafa",
      "flex": 1,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": undefined,
      }
    }
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "margin": 6,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 26.25,
            "height": 52.5,
            "width": 52.5,
          },
          undefined,
          Object {
            "alignSelf": "flex-end",
            "marginRight": 10,
            "marginTop": 40,
          },
        ],
      ]
    }
  >
    <View>
      <Text />
    </View>
  </View>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 30,
        "paddingBottom": 10,
      }
    }
  >
     Preview 
  </Text>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fafafa",
        "borderWidth": 1,
        "flex": 1,
        "height": 0,
        "paddingHorizontal": 50,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      placeholder="ABC company"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "fontSize": 30,
          "marginTop": 20,
          "paddingBottom": 10,
        }
      }
      underlineColorAndroid="transparent"
    />
    <Image
      source={1}
      style={
        Object {
          "alignSelf": "center",
          "borderWidth": 1,
          "height": 250,
          "marginBottom": 100,
          "marginTop": 50,
          "resizeMode": "contain",
          "width": 250,
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      multiline={true}
      numberOfLines={5}
      placeholder="Edit Message: Scan the QR code to check into our business."
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "borderBottomWidth": 1,
          "fontSize": 20,
          "paddingBottom": 10,
          "width": 300,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#0a0540",
        "borderRadius": 25,
        "marginBottom": 40,
        "opacity": 1,
        "padding": 15,
        "width": 150,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fafafa",
        }
      }
    >
      Save
    </Text>
  </View>
</View>
`;
